app/lib/shared/deps_graph.dart

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:gcloud/db.dart';
+import 'package:logging/logging.dart';
+
+import 'package:pub_dartlang_org/frontend/models.dart';
+import 'package:pub_dartlang_org/shared/task_scheduler.dart';
+
+final Logger _logger = new Logger('pub.package_graph');
+
+// To avoid having the same package names many times in memory we canonicalize them.
+class StringCanonicalization {
```

Comment:
The name gave me a pause, as I was expecting some kind of text transformation or an URI something that is an encoded form of the package name.

I'm trying to figure out a better name for it (around the expressions of singleton, dictionary related phrases), but haven't found a good one yet.

Reply:
Found the correct term: https://en.wikipedia.org/wiki/String_interning

Reply:
That is a nice find, Istvan. Never heard of "string interning" before. But it's better to use a term that almost everyone is familiar with and yes, [canonicalization](https://en.wikipedia.org/wiki/Canonicalization) is a very good name for it.

You can try searching for "canonicalization" and "interning" in the dart sdk and see which is more common.